### PR TITLE
Add database restore capability

### PR DESCRIPTION
**Pull request by merrilmathew**
### Context

Postgres database for Find A Lost TRN has point in time backups available. This PR adds changes to Makefile and Terraform configuration to run the restore process from a local machine. This enables developers to recover data in disaster recovery.

### Trello card

https://trello.com/c/e1ay9JO0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,6 +22,11 @@ read-keyvault-config:
 	$(eval KEY_VAULT_NAME=$(shell jq -r '.key_vault_name' terraform/workspace_variables/$(DEPLOY_ENV).tfvars.json))
 	$(eval KEY_VAULT_SECRET_NAME=$(shell jq -r '.key_vault_secret_name' terraform/workspace_variables/$(DEPLOY_ENV).tfvars.json))
 
+read-deployment-config:
+	$(eval SPACE=$(shell jq -r '.paas_space' terraform/workspace_variables/$(DEPLOY_ENV).tfvars.json))
+	$(eval POSTGRES_DATABASE_NAME=$(shell jq -r '.postgres_database_name' terraform/workspace_variables/$(DEPLOY_ENV).tfvars.json))
+	$(eval FLT_APP_NAME=$(shell jq -r '.flt_app_name' terraform//workspace_variables/$(DEPLOY_ENV).tfvars.json))
```
Comment (saliceti) [inline Makefile:28]: Double //

File: Makefile [hunk 2]
```diff
@@ -45,6 +54,44 @@ validate-infra-secrets: read-keyvault-config install-fetch-config set-azure-acco
...
+rename-postgres-service: read-deployment-config ## make dev rename-postgres-service NEW_NAME_SUFFIX=old CONFIRM_RENAME
+	$(if $(CONFIRM_RENAME), , $(error can only run with CONFIRM_RENAME))
+	$(if $(NEW_NAME_SUFFIX), , $(error NEW_NAME_SUFFIX is required))
+	cf target -s ${SPACE} > /dev/null
+	cf rename-service  ${POSTGRES_DATABASE_NAME} ${POSTGRES_DATABASE_NAME}-$(NEW_NAME_SUFFIX)
+
+remove-postgres-tf-state: terraform-init ## make dev remove-postgres-tf-state PASSCODE=XXX
+	cd terraform && terraform state rm cloudfoundry_service_instance.postgres
+
+restore-postgres: terraform-init read-deployment-config ## make dev restore-postgres DB_INSTANCE_GUID="<cf service db-name --guid>" BEFORE_TIME="yyyy-MM-dd hh:mm:ss" TF_VAR_api_docker_image=ghcr.io/dfe-digital/qualified-teachers-api:<COMMIT_SHA> PASSCODE=<auth code from https://login.london.cloud.service.gov.uk/passcode>
```
Comment (saliceti) [inline Makefile:86]: qualified-teachers-api

File: docs/restore-paas-postgres-database.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+Login to Azure using Azure CLI. Run `az login` if not logged in previously. This will take you to the azure portal to authenticate the cli. This is required for reading the terraform state and KeyVault secrets.
+
+### 4. Rename the postgres database
+
+This is because as part of the recovery process we will recreate the service using the same name as before with the backup parameters in one terraform action.
+
+Run `make <ENV> rename-postgres-service CONFIRM_RENAME=1 NEW_NAME_SUFFIX=old`
+
+### 5. Remove existing database service from terrform state
```
Comment (saliceti) [inline docs/restore-paas-postgres-database.md:52]: terrform

**Comment by merrilmathew** [comment]
[Add postgres database restore option](https://trello.com/c/e1ay9JO0/386-add-postgres-database-restore-option)

**Review by saliceti — APPROVED** [review]

**Review by RMcVelia — APPROVED** [review]

File: docs/restore-paas-postgres-database.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+Review the generated deploy plan and check that the database instance will be created using the specified point in time backup.
+Make sure there are no changes to the **flt_docker_image** property of the app as we are deploying the app again with the same version as before the incident.
+
+Once the plan is reviewed, type **yes** to initiate the deployment process. This process can take between 10 and 60 mins to complete. When this is in progress, you can run cf service <db-instance-name> to check the progress. The database service instance will be created and the dependant app will be started on successful completion.
+
+### 7. Delete renamed database instance
+
+Once restore has been ran successfully and checked, make sure to delete the renamed instance created before the restore process was intitated in step 4.
+
+Run `cf delete-service <db-service-instance-name>-old`
```
Comment (tvararu) [inline docs/restore-paas-postgres-database.md:85]: Really complete looking docs :+1:

**Pull request merged**